gammapy/cube/simulate.py

```diff
@@ -109,6 +109,49 @@ class MapDatasetEventSampler:
     def __init__(self, random_state="random-seed"):
         self.random_state = get_random_state(random_state)
 
+    def sample_sources(self, dataset):
+        """Sample source model components.
+        Parameters
+        ----------
+        dataset : `MapDataset`
+            Map dataset.
+
+        Returns
+        -------
+        events : `EventList`
+            Event list
+        """
+        events_all = []
+        for idx, evaluator in enumerate(dataset._evaluators):
+            table = Table()
+            evaluator.edisp = None
+            evaluator.psf = None
+            npred = evaluator.compute_npred()
+            n_events = self.random_state.poisson(np.sum(npred.data))
+
+            # sample position
+            coords = npred.sample_coord(n_events, self.random_state)
+            table["ENERGY"] = coords["energy"]
```

Comment:
Note that this column should be `"ENERGY_TRUE"` and the same for `"RA_TRUE"` and `"DEC_TRUE"`